main.rb

```diff
@@ -0,0 +1,59 @@
+require 'open3'
+require 'os'
+
+def get_env_variable(key)
+  return (ENV[key] == nil || ENV[key] == "") ? nil : ENV[key]
+end
+
+def env_has_key(key)
+  value = get_env_variable(key)
+	return value unless value.nil? || value.empty?
+
+	abort("Input #{key} is missing.")
+end
+
+$output_path = env_has_key("AC_OUTPUT_DIR")
+$repo_path = env_has_key("AC_REPOSITORY_DIR")
+$detox_configuration = env_has_key("AC_RN_DETOX_CONFIGURATION")
+
+def run_command(command)
+  puts "@@[command] #{command}"
+  stdout_str, stderr_str, status = Open3.capture3(command)
+
+  if status.success?
+    puts stdout_str unless stdout_str.empty?
+    puts stderr_str unless stderr_str.empty?
+    return stderr_str.empty? ? stdout_str : stderr_str
+  else
+    return stderr_str
+  end
+end
+
+def runTests
+
+    if OS.mac?
+      run_command("brew tap wix/brew && brew install applesimutils")
+    end 
+
+    yarn_or_npm = "npm"
+    if File.file?("#{$repo_path}/yarn.lock")
+      yarn_or_npm = "yarn"
+    end
+
+    puts "Detox Configuration used is #{$detox_configuration}"
+
+    run_command("cd #{$repo_path} && #{yarn_or_npm} detox clean-framework-cache")
+    run_command("cd #{$repo_path} && #{yarn_or_npm} detox build-framework-cache --configuration #{$detox_configuration}")
+    run_command("cd #{$repo_path} && #{yarn_or_npm} detox build --configuration #{$detox_configuration}")
+    run_command("cd #{$repo_path} && #{yarn_or_npm} detox test --configuration #{$detox_configuration} --take-screenshots all")
```

Comment:
@csonuryilmaz There are other options for "detox", do you think we should add extra args inputs for those?

- for [detox test](https://wix.github.io/Detox/docs/19.x/api/detox-cli/#test)
- for [detox build](https://wix.github.io/Detox/docs/19.x/api/detox-cli/#build)

Reply:
@ftmzhrsckldr To be able to run test, first project should build first. For example in the other test components like ios unit and ui tests, it builds project first and run the tests @csonuryilmaz

Reply:
> ... it builds project first and run the tests

OK, no problem with that behavior.

> ... other options for "detox", do you think we should add extra args ...

No need for the `build` command.  But the `test` command has some useful functionality from which the user can benefit.

It will be good to add an optional args input (freeform textbox) for the `test` command in this PR.
- detox [test](https://wix.github.io/Detox/docs/19.x/api/detox-cli/#test)

It might not be possible to support all the options, but we can test a subset of them in our component testing process.

Reply:
```ruby
run_command("cd #{$repo_path} && #{yarn_or_npm} detox test --configuration #{$detox_configuration}")
```

☝️ The `test` command can be like the above.

We can set the default value of optional args as: ` --take-screenshots all`